src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -126,6 +170,27 @@ private static void initHttpClient() {
     httpClient = clientBuilder.build();
   }
 
+  /** Starts a daemon thread to monitor idle connections in http connection manager. */
+  private static void initIdleConnectionMonitoringThread() {
+    idleConnectionMonitorThreadLock.lock();
+    try {
+      // Start a new thread only if connectionManager was init before and
+      // daemon thread was not started before or was closed because of SimpleIngestManager close
+      if (connectionManager != null
+          && (idleConnectionMonitorThread == null || idleConnectionMonitorThread.isShutdown())) {
+        // Monitors in a separate thread where it closes any idle connections
+        // https://hc.apache.org/httpcomponents-client-4.5.x/current/tutorial/html/connmgmt.html
+        idleConnectionMonitorThread = new IdleConnectionMonitorThread(connectionManager);
+        idleConnectionMonitorThread.setDaemon(true);
+        idleConnectionMonitorThread.start();
+      }
+    } catch (Exception e) {
+      LOGGER.warn("Unable to start Daemon thread for Http Idle Connection Monitoring", e);
```

Comment:
logger.error?

Reply:
Didnt want to throw this as error, as this is more for debugging. warn should be fine.